.vib/victoriametrics/cypress/cypress/e2e/victoriametrics.cy.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Broadcom, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: APACHE-2.0
+ */
+
+/// <reference types="cypress" />
+import { random } from '../support/utils';
+
+// All tests use vmauth as the access
+
+// This verifies that vmui and vmagent work
+// vmagent pushes metrics, and we use vmui to query them
+it('Access VMUI and check metric ingested by vmagent (access via vmauth)', () => {
+  cy.visit('/select/0/vmui');
+  cy.fixture('vmagent_metric').then((metric) => {
+    cy.get('[inputmode="search"]').type(metric.name);
+    cy.contains('Execute Query').click();
+    cy.contains('JSON').click();
+    cy.contains(metric.expectedResult);
+  });
+});
+
+// This verifies that vminsert, vmselect and vmstorage work
+// With vminsert we push a metric, which gets stored in vmstorage. Then it is
+// queried using vmselect
+it('Push metric using the vminsert API and then query using vmselect (all via vmauth)', () => {
+  cy.fixture('ingest_metric').then((metric) => {
+    // Metrics need to start with a letter
+    const ingestPromQL = `a${random}${metric.ingestPromQL}`;
+    const queryPromQL = `a${random}${metric.queryPromQL}`;
+    cy.request({
+      url: '/insert/0/prometheus/api/v1/import/prometheus',
+      method: 'POST',
+      body: ingestPromQL,
+    })
+      .its('status')
+      .should('be.equal', 204)
+      .then(() => {
+        // It is not instantly available for querying, so we need to wait
+        cy.wait(45000);
```

Comment:
I guess the availability depends on some "scrape interval" value (or similar). It'd be nice if we could set this based on that instead of hardcoding 45 seconds.

Reply:
Update: Used that value (verified in configuration) but it didn't work. Leaving it as it is